src/ophyd_async/core/_derived_signal.py

```diff
@@ -200,7 +207,7 @@ def derived_signal_r(
     raw_to_derived: Callable[..., SignalDatatypeT],
     derived_units: str | None = None,
     derived_precision: int | None = None,
-    **raw_devices: Device,
+    **raw_devices_and_constants: Device | int | bool | float | str,
```

Comment:
Could: there is a union of these primitive types already in `._signal_backend`

```suggestion
    **raw_devices_and_constants: Device | Primitive,
```